src/argilla/server/daos/backend/search/model.py

```diff
@@ -74,6 +74,13 @@ class BaseRecordsQuery(BaseQuery):
 
     has_annotation: Optional[bool] = None
     has_prediction: Optional[bool] = None
+    # TODO(@ufukhurriyetogl): In order to allow embedding-based searches, we can include here
+    #  (the base records query data model class) an embeddings related query params that
+    #  we can use later for the records search.
+    # The input vector:
+    embedding_vector: Optional[List[float]] = None
+    # The name of the embedding to uses (if several):
+    embedding_name: Optional[str] = None
```

Comment:
Here, which name should be provided to match the stored one? Regarding the data model discussed yesterday, given the embedding config for a record:

```python
embeddings = {
        "text": {
            "embedding_vectors": { 
                "bert_uncased": [...],
                "bert_case": [...]
            }
        },
        "inputs": {
            "property_names": ["body", "subject"],
            "embedding_vectors": { 
                "bert_uncased": [...],
                "bert_case": [...]
            }
        }
}
```
If a user wants to filter using the `bert_case` for the input text, must compound the final embedding vector like the server does, which is not desirable.

Also, for several embedding based on different fields from the input, this data model does not match (since we have only one `inputs` key)


I think it should be better if we let the users to naming on their own embeddings, so those names can be used easily in query time. For example:

```python
embeddings = {
   "my_text_bert_cased": { 
       "vector": [...],
       # this field is not needed for now, and we could avoid it for the moment. 
       # Once we introduce automatic embedding generation we can refine how 
       # to choice the record field used for the embeding
       "record_properties": ["inputs.body", "inputs.subject"] 
   }, 

   "another_vector_uncased": {
      "vector": [...],
      "record_properties": [ { "inputs": ["body", "subject"]}, "text"] # Another definition way 
      "model": "bert_uncased" # Once we have automatic embedding generation, we can provide the model name used for that in this way
   }
}
```

So, under this configuration, we can easily query data using embedding likes follows:

````python
embedding_name = "another_vector_uncased"
embedding_vector = [...]
````

What do you think @ufukhurriyetoglu ?